### PR TITLE
Magento 117

### PR DESCRIPTION
**Pull request by suraiyavic**
Would like to test some edge cases and different work flows before setting it to regular PR.

### CODE REVIEW

File: Model/DutyCalculator.php [hunk 1]
```diff
@@ -288,7 +290,6 @@ public function fillOutQuoteAndSessionUsingFeeReturned($duty, $address, $apply,
             $this->checkoutSession->setBaseReachDuty(0);
             $this->checkoutSession->setReachDuty($duty_adjusted);
             $quote->setDhlQuoteId('');
-            $this->checkoutSession->setDhlQuoteId('');
```
Comment (lizzyWithReach) [inline Model/DutyCalculator.php:291]: if the session isn't getting cleared when the checkbox isn't selected, then couldn't a customer 

- go through the checkout flow
- click the duties and taxes checkbox (the session gets a quoteId), 
- leave the checkout flow then when they 
- go through checkout again they 
- don't click the checkbox/deselect checkbox, 
- wouldn't their session still have the old quoteId?

Just want to double check that removing the clearing of the quoteId in the session doesn't result in issues like the one I outlined above

File: Model/DutyCalculator.php [hunk 2]
```diff
@@ -278,6 +279,7 @@ public function fillOutQuoteAndSessionUsingFeeReturned($duty, $address, $apply,
             $quote->setDhlBreakdown(json_encode($response['feeTotals']));
             $this->checkoutSession->setDhlBreakdown($quote->getDhlBreakdown());
             $this->checkoutSession->setApply(true); //checkbox selection /corresponding passed value
+            $this->checkoutSession->setDhlQuoteId($response['quoteId']);
```
Comment (lizzyWithReach) [inline Model/DutyCalculator.php]: The comment on line 260 says: `//DHL quoteID is already saved into session; so not doing it here again` , if we want to save the quote into the session again (which I don't mind, it should be the same value and if not then this is a good chance to correct that) could you delete this comment if it makes sense to do so?

**Review by lizzyWithReach — CHANGES_REQUESTED** [review]
I could use a bit of clarification and I have a comment removal suggestion, thank you!

File: Model/DutyCalculator.php (hunk 1 above)
Comment (suraiyavic) [reply Model/DutyCalculator.php:291]: Few things Lizzy:

Some values are not getting cleared from session (deliberately). This is to be able to reuse the D&T (as well as DHLQuoteid) values without a new DHL call (if nothing else changes except the checkbox selection ).  
What is getting into the database is from Quote not from Session.
So when user is deselecting the apply D&T checkbox - the info related to D&T is getting cleared from the Quote but not from Session. This is so that just in case the user select the apply D&T option again we can reload value from session to quote without making a new DHL call. If we clear the DHLQuoteID from session then if the user changes mind during the same session then we would not have the DHLquoteid without making a new call.
However there are cases when it would be a problem (like as listed in MAG-115). This is the reason I was requesting (please have a look at `dev-magento` slack channel ) last Friday to bump up the priority of MAG-115 so that more event handlers (like MAG-81) are added for proper handling of such cases. However that did not happen (I am going to share the link with you shortly). Thanks!


 @lizzynibali

File: Model/DutyCalculator.php (hunk 2 above)
Comment (suraiyavic) [reply Model/DutyCalculator.php]: Hello Lizzy, Shall have a look/cleanup once my system is not `in transition` state. Right now updating few things on my dev setup. Thanks!

File: Model/DutyCalculator.php (hunk 1 above)
Comment (lizzyWithReach) [reply Model/DutyCalculator.php:291]: ohh thank you! I appreciate the time you spent clarifying this

File: Model/DutyCalculator.php (hunk 2 above)
Comment (lizzyWithReach) [reply Model/DutyCalculator.php]: sounds good! take your time

Comment (suraiyavic) [reply Model/DutyCalculator.php]:
```
//DHL quoteID is already saved into session; so not doing it here again
        $quote = $this->checkoutSession->getQuote();
```
I can remove the comment. Thanks!

Comment (suraiyavic) [reply Model/DutyCalculator.php]: commit 15153627c4dedfebbb8417849bfb680a56b6e1f5 
Thank you. 
On  merging this back to magento-25 branch (as that is cut from this magento-117 branch)

Comment (lizzyWithReach) [reply Model/DutyCalculator.php]: yay! thanks!

**Review by lizzyWithReach — APPROVED** [review]
